Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/MaterialBrowserInteractions.cpp

```diff
@@ -39,6 +46,61 @@ namespace AZ
             }
         }
 
+        void MaterialBrowserInteractions::AddSourceFileCreators(
+            [[maybe_unused]] const char* fullSourceFolderName,
+            [[maybe_unused]] const AZ::Uuid& sourceUUID,
+            AzToolsFramework::AssetBrowser::SourceFileCreatorList& creators)
+        {
+            using namespace AtomToolsFramework;
+
+            creators.push_back(
+                { "Material_Creator", "Material", QIcon(),
+                  [&]([[maybe_unused]] const char* fullSourceFolderNameInCallback, [[maybe_unused]] const AZ::Uuid& sourceUUID)
+                  {
+                      const AZ::Data::AssetId assetId = AtomToolsFramework::GetSettingsObject<AZ::Data::AssetId>(
+                          "/O3DE/Atom/MaterialEditor/DefaultMaterialTypeAsset",
+                          AZ::RPI::AssetUtils::GetAssetIdForProductPath("materials/types/standardpbr.azmaterialtype"));
+                      QWidget* mainWindow = nullptr;
+                      AzToolsFramework::EditorRequests::Bus::BroadcastResult(
+                          mainWindow, &AzToolsFramework::EditorRequests::Bus::Events::GetMainWindow);
+
+                      CreateDocumentDialog dialog(
+                          QObject::tr("Create Material"), QObject::tr("Select Type"), QObject::tr("Select Material Path"),
+                          fullSourceFolderNameInCallback, { "material" }, assetId,
+                          []([[maybe_unused]] const AZ::Data::AssetInfo& assetInfo)
+                          {
+                              AZStd::unordered_set<AZ::Uuid> supportedAssetTypesToCreate;
+                              supportedAssetTypesToCreate.insert(azrtti_typeid<AZ::RPI::MaterialTypeAsset>());
+
+                              // If any asset types are specified, do early rejection tests to avoid expensive string comparisons
+                              const auto& assetTypes = supportedAssetTypesToCreate;
+                              if (assetTypes.empty() || assetTypes.find(assetInfo.m_assetType) != assetTypes.end())
+                              {
+                                  // Additional filtering will be done against the path to the source file for this asset
+                                  const auto& sourcePath = AZ::RPI::AssetUtils::GetSourcePathByAssetId(assetInfo.m_assetId);
+
+                                  // Only add source files with extensions supported by the document types creation rules
+                                  // Ignore any files that are marked as non editable in the registry
+                                  return AZ::StringFunc::EndsWith(sourcePath.c_str(), AZ::RPI::MaterialTypeSourceData::Extension) &&
+                                      IsDocumentPathEditable(sourcePath);
+                              }
+                              return false;
```

Comment:
This is just doing an asset type filter so the drop down only shows asset types we care about. The material editor supports multiple types to create a new document from. Here, you only need to check for one type. There's also no need to check with the extension because there's only one type. in this case comma you should be able to shrink it to one line.

```suggestion
return assetInfo.m_assetType) == azrtti_typeid<AZ::RPI::MaterialTypeAsset>() && IsDocumentPathEditable(AZ::RPI::AssetUtils::GetSourcePathByAssetId(assetInfo.m_assetId));
```

Reply:
Nice. Change made.